Incorporate new Yield Director contract into the Give system

0xLienid opened the pull request.

sam-potter committed:
Abstract switch network component

sam-potter committed:
Update test snapshot

0xJem committed:
Shift to using the topLeft prop of the Paper component and Grid

0xJem committed:
Fix sizing of grant image in medium breakpoint

0xJem committed:
Correct sizing of milestone details

0xJem committed:
Add new icons (though they have theme problems)

0xJem committed:
Correct sizing of SVG icons

0xJem committed:
Merge remote-tracking branch 'origin/develop' into tyche-grants

0xJem committed:
Update failing snapshot test

0xJem committed:
Remove redundant code

sam-potter committed:
Convert mutations

sam-potter committed:
Update test snapshots

sam-potter committed:
Merge branch 'develop' into react-query/wrap-view

0xJem committed:
Align grant profile with design 2.0.2

0xJem committed:
Add the ability to hide a grant from display

0xJem committed:
Update grants milestone amounts

0xJem committed:
Fix rendering of the donors icon with themes. This required explicitly adding a fill property, and defining the stroke property as currentcolor

0xJem committed:
Workaround for SVG fills

0xJem committed:
Add remaining 2 SVG icons for Grants (with stroke/fill fixes)

0xJem committed:
Fix wrapping of donors

0xJem committed:
Remove redundant grant

0xJem committed:
Didn't mean to enable that

sam-potter committed:
Abstract WalletConnectedGuard

sam-potter committed:
Convert cross chain migration

sam-potter committed:
Cleanup

0xJem committed:
Correct layout on md and lg breakpoints

sam-potter committed:
Update test snapshots

sam-potter committed:
Merge branch 'develop' into react-query/wrap-view

sam-potter committed:
Stop making unecessary requests to testnets

sam-potter committed:
Add react query error handler

0xJem committed:
Fix SVG rendering of fills

0xJem committed:
SVG updates
- Add new/updated icons
- Remove redundant icons
- Fix SVG background/fill issues

These SVG icons will be moved into component-library shortly.

0xJem committed:
Update EducationCard with latest SVG icons

0xJem committed:
Merge remote-tracking branch 'origin/develop' into tyche-grants

0xJem committed:
Hide project stats for grants

0xJem committed:
Merge remote-tracking branch 'origin/HEAD' into tyche-grants

0xJem committed:
Remove extraneous log

0xJem committed:
Update to component-library 1.7.1

0xJem committed:
Shift icons in GrantCard to component-library

0xJem committed:
Migrate ProjectCard to component-library icons

0xJem committed:
Shift EducationCard to use component-library icons

0xJem committed:
Remove redundant SVG files

0xJem committed:
Add feature flag to disable Give Grants

0xJem committed:
Misc design fixes

0xJem committed:
Update snapshots

0xJem committed:
Disable individual grant projects until they are approved. Also display helpful text if no grants are active.

0xJem committed:
Redundant Grid component

0xJem committed:
Remove redundant CSS

0xJem committed:
Shift CauseCard to use Grid component

0xJem committed:
WIP refactoring of ProjectCard to use Grid component

0xJem committed:
Remove redundant className

0xJem committed:
Remove redundancies

0xJem committed:
Revert temporary change to user donating flag

0xJem committed:
Remove redundant CSS

0xJem committed:
Remove redundant CSS

0xJem committed:
Update test snapshot with recent changes

0xJem committed:
Remove redundant CSS

0xJem committed:
Simplify the image used in Project and Grant screens

0xJem committed:
Convert amount screen of RecipientModal to use the Grid component

0xJem committed:
WIP conversion of RecipientModal to Grid component

0xLienid committed:
Code style and documentation

0xLienid committed:
Fix merge conflict

0xJem commented on src/helpers/GetDonationDate.ts:
```diff
@@ -4,41 +4,46 @@ import { addresses } from "../constants";
-    const dateString = month + " " + day + ", " + year;
-
-    return dateString;
+  if (!(addresses[networkID] && addresses[networkID].GIVING_ADDRESS)) {
```
This approach is good, but could be better. I would suggest:
```
if (!addresses[networkID] || !addresses[networkID].GIVING_ADDRESS) {
   <etc…>
}

<rest of the code>
```

No need for the else statement.

0xJem replied:
Why? 
- It prevents having lots of nested if/else statements.
- The conditions are easier to understand: not A or not B, instead of not A and not B.

0xJem commented on src/helpers/GetDonationDate.ts:
```diff
@@ -4,41 +4,46 @@ import { addresses } from "../constants";
+    const events = await provider.getLogs(filter);
+
+    // Check that there are any events
+    if (events.length > 0) {
```
Probably do the same here. If you can exit early, do so.

If events.length === 0, return “”

0xJem commented on src/helpers/GetTotalDonated.ts:
```diff
@@ -32,13 +38,70 @@ export const getTotalDonated = async ({ address, networkID, provider }: IBaseAdd
+  }
+};
+
+// Calculate total yield sent from a given user to a given recipient
```
Do the `/**` style function comment, as VSCode (etc) will automatically recognise it, so that when you hover over the function call in another file, it’ll display the documentation.

0xJem commented on src/slices/AccountSlice.ts:
```diff
@@ -286,34 +288,48 @@ export const getDonationBalances = createAsyncThunk(
+            });
+
+            // NOTE: Bad fix, but since no rebases on testnet this would throw an error otherwise
+            const yieldSent: BigNumber =
```
Given that we check for `addresses[networkID]` in getTotalYieldSent and return a zero BigNumber if not found, isn’t this check redundant?

0xLienid replied:
I don't think so, since addresses[networkID] and addresses[networkID].GIVING_ADDRESS both exist on testnet but rebases don't, so this makes sure that we're on mainnet otherwise sets it to 0. I can move this to that check for now if you'd like though.

0xLienid replied:
moved to getTotalYieldSent

0xJem commented on src/slices/AccountSlice.ts:
```diff
@@ -286,34 +288,48 @@ export const getDonationBalances = createAsyncThunk(
+                    provider: provider,
+                  })
+                : BigNumber.from("0");
+            const sohmYieldSent = await gohmContract.balanceFrom(yieldSent);
```
There are two async calls here, within a for loop. As the number of deposits increases, isn’t this going to become a performance bottleneck? You may want to look at batching the async calls. If out of scope for this PR/sprint, then log a task.

0xJem commented on src/slices/AccountSlice.ts:
```diff
@@ -368,6 +384,7 @@ export const getMockDonationBalances = createAsyncThunk(
             const formattedYieldSent = ethers.utils.formatUnits(yieldSent, "gwei");
 
             donationInfo.push({
+              id: "1",
```
Document the reason why the id is 1

0xJem commented on src/views/Give/CausesDashboard.tsx:
```diff
@@ -74,6 +74,7 @@ export default function CausesDashboard() {
           action: ACTION_GIVE,
           value: depositAmount.toFixed(),
           recipient: walletAddress,
+          id: "-1",
```
What does `-1` indicate here?

0xJem commented on src/views/Give/WithdrawDepositModal.tsx:
```diff
@@ -0,0 +1,113 @@
+    /* modal-container displays a background behind the ohm-card container, which means that if modal-container receives a click, we can close the modal */
+    <Modal className="modal-container" open={isModalOpen} onClose={cancelFunc} onClick={cancelFunc} hideBackdrop={true}>
+      <Paper className="ohm-card ohm-modal" onClick={handleModalInsideClick}>
+        <div className="yield-header">
```
I would do this from scratch using Grid components. Let me know if you have time to wrap your head around that!

0xJem committed:
Fix breakpoints on confirmation screen.

0xJem committed:
Align RecipientModal with reference design

0xJem committed:
Migrate GiveInfo. Remove redundant CSS.

0xJem committed:
Update snapshot

0xJem committed:
Shift CausesDashboard to use MUI components

0xJem committed:
Cleanup Give component

0xJem committed:
Update snapshot

0xJem committed:
Migrate CausesDashboard and GrantsDashboard

0xJem committed:
Clean up redundant CSS (LOTS of it)

0xJem committed:
Fix breakpoints for project and grant card

0xJem committed:
Responsive version of no donations message

0xJem committed:
Migrate manage donation screen

0xJem committed:
Migrate ManageDonationModal

0xJem committed:
Shift to use GiveBox

0xJem committed:
Convert RedeemYield to use Grid. Remove redundant CSS.

0xJem committed:
Use the DataRow component

0xJem committed:
Remove redundant component

0xJem committed:
Adjust breakpoint

0xJem committed:
Accidental commit

0xJem committed:
Migrate RedeemYieldModal to use Grid. Cleanup CSS.

0xJem committed:
Revert feature flag for give grants (since we will display an informative message instead)

sam-potter committed:
Merge branch 'develop' into react-query/wrap-view

sam-potter committed:
Update import

sam-potter committed:
Update test snapshots

sam-potter committed:
Move file

sam-potter committed:
Add react-query devtools

sam-potter committed:
Add error handler

0xLienid committed:
Code style and batched async calls

0xJem committed:
Merge branch 'develop' into tyche-grants

# Conflicts:
#	.env.example
#	.env.test
#	src/locales/translations

0xJem committed:
Compilation fixes due to merge

0xJem committed:
Rebase fix

0xJem committed:
Layout fixes. Remove custom styling where possible.

0xJem committed:
Display message if there is no milestone information

0xJem committed:
Clean up EducationCard: shift to Grid components, remove redundant graphics, rename components.

0xJem committed:
Remove redundant SVG fill

0xLienid committed:
Comments and documentation

0xLienid committed:
Tweaks to data updating

0xLienid committed:
New contract address and ABI

sam-potter committed:
Add support for migrating specific amounts

sam-potter committed:
Refactor

sam-potter committed:
Remove unused migrate functions

sam-potter committed:
Fix translations

sam-potter committed:
Merge branch 'develop' into react-query/wrap-view

sam-potter committed:
Use testnets

sam-potter committed:
Prefer global query error handler

sam-potter committed:
Fix incorrect boolean

sam-potter committed:
Use testnets

0xJem commented on src/components/GiveProject/GrantCard.tsx:
```diff
@@ -76,7 +76,7 @@ export default function GrantCard({ grant, mode }: GrantDetailsProps) {
   const [donorCount, setDonorCount] = useState(0);
   const [isUserDonating, setIsUserDonating] = useState(false);
-  const [donationId, setDonationId] = useState(0);
+  const [donationId, setDonationId] = useState(-1);
```
Document the reason for this initial value

0xJem commented on src/components/GiveProject/GrantCard.tsx:
```diff
@@ -96,6 +96,8 @@ export default function GrantCard({ grant, mode }: GrantDetailsProps) {
+  // Resets the viewport to the top of the page when pathnames change rather than
+  // preserving vertical position of the page you are coming from
   useEffect(() => {
     const items = document.getElementsByClassName("project-container");
```
We're doing away with all of the custom classes, so it's better that you use an `id` attribute to refer to that particular element.

0xLienid replied:
changed, but not sure if it's what you meant so probably would be good to check it again when I re-push

0xJem commented on src/helpers/GetDonationDate.ts:
```diff
@@ -3,7 +3,20 @@ import { ethers } from "ethers";
 import { addresses } from "../constants";
 import { IBaseAddressRecipientAsyncThunk } from "../slices/interfaces";
 
+/**
```
*chef's kiss*

0xJem commented on src/helpers/GetDonationDate.ts:
```diff
@@ -3,7 +3,20 @@ import { ethers } from "ethers";
+ * @param provider Network provider object
+ * @returns String representation of the date as Month Day, Year
+ */
 export const GetDonationDate = async ({ address, recipient, networkID, provider }: IBaseAddressRecipientAsyncThunk) => {
```
A better name would be `GetFirstDonationDate`

0xJem commented on src/helpers/GetDonationDate.ts:
```diff
@@ -3,7 +3,20 @@ import { ethers } from "ethers";
+ * @param recipient The donation target of the current user
+ * @param networkID ID number of the network the user is currently connected to
+ * @param provider Network provider object
+ * @returns String representation of the date as Month Day, Year
```
Why return this as a string? Why not a native Date?

If you want to return it as a string, document the format that is used.

0xLienid replied:
It felt like the simplest route to convert it to a good display format if the same display format was acceptable everywhere we would be using it. But, I'm not married to doing it that way. Do you think it's better to return as a Date and convert at the point of use?

0xLienid replied:
For now just documenting format

0xJem commented on src/helpers/GetTotalDonated.ts:
```diff
@@ -1,44 +1,122 @@
-import { IBaseAddressAsyncThunk } from "../slices/interfaces";
+import { IBaseAddressAsyncThunk, IBaseAddressRecipientAsyncThunk } from "../slices/interfaces";
 
 // Calculate total amount redeemed by a user + their current redeemable balance
```
Extraneous line

0xJem commented on src/helpers/GetTotalDonated.ts:
```diff
@@ -1,44 +1,122 @@
-    }
+  if (!addresses[networkID] || !addresses[networkID].GIVING_ADDRESS) {
+    console.log("No giving contract on chain ID " + networkID);
+    return "0";
```
This is returning a string, not a BigNumber.

0xJem commented on src/helpers/GetTotalDonated.ts:
```diff
@@ -1,44 +1,122 @@
+ * @returns BigNumber representation of the sOHM value of the total amount redeemed by a user
+ *          plus their current redeemable balance
+ */
 export const getTotalDonated = async ({ address, networkID, provider }: IBaseAddressAsyncThunk) => {
```
I would suggest being explicit about the return type, so that it catches any deviations (e.g. line 20 returning "0" instead of BigNumber).

0xLienid replied:
Made this one return a string,

0xJem commented on src/helpers/GetTotalDonated.ts:
```diff
@@ -1,44 +1,122 @@
+  const totalDonated = totalRedeemed.add(redeemableBalance);
+  const totalDonatedAsSohm = await gohmContract.balanceFrom(totalDonated);
+
+  return ethers.utils.formatUnits(totalDonatedAsSohm, "gwei");
```
This contradicts line 14, which says that a BigNumber is returned.

0xJem commented on src/helpers/GetTotalDonated.ts:
```diff
@@ -1,44 +1,122 @@
+  recipient,
+  networkID,
+  provider,
+}: IBaseAddressRecipientAsyncThunk) => {
```
Same suggestion: explicit return type.

0xJem commented on src/helpers/GiveRedemptionBalanceHelper.ts:
```diff
@@ -18,33 +17,58 @@ interface IDonorAddresses {
+ * @param provider Ethereum network provider object
+ * @returns Redeemable sOHM plus recipientInfo object (totalDebt and agnosticDebt)
+ */
 export const getRedemptionBalancesAsync = async ({ address, networkID, provider }: IBaseAddressAsyncThunk) => {
```
Wouldn't hurt to be explicit about the return type.

0xJem commented on src/helpers/GiveRedemptionBalanceHelper.ts:
```diff
@@ -18,33 +17,58 @@ interface IDonorAddresses {
   };
 
-  if (addresses[networkID] && addresses[networkID].GIVING_ADDRESS) {
+  if (!(addresses[networkID] && addresses[networkID].GIVING_ADDRESS)) {
```
I suggest exiting the function here.

0xLienid replied:
it does doesn't it? If that condition is hit, console.log's the error message and then jumps straight to the return.

0xJem replied:
Cleaner to just exit then and there.

0xJem commented on src/helpers/GiveRedemptionBalanceHelper.ts:
```diff
@@ -18,33 +17,58 @@ interface IDonorAddresses {
-        recipientInfoData.indexAtLastChange.toNumber(),
-        "gwei",
-      );
+      const recipientIds = await givingContract.getRecipientIds(address);
```
Why is this in try/catch but the `givingContract` and `gohmContract` calls aren't?

0xLienid replied:
because we've already determined that givingContract exists, so there's no concern about instantiating it, and totalRedeemableBalance can't revert. getRecipientIds can return a null value though which then could create problems in future calls. I don't mind adding everything in the try/catch if you think it makes the code neater to look at/consistent

0xJem commented on src/helpers/GiveRedemptionBalanceHelper.ts:
```diff
@@ -97,50 +114,60 @@ export const getMockRedemptionBalancesAsync = async ({ address, networkID, provi
-  };
+  const donationsToAddress = [];
+
+  if (!(addresses[networkID] && addresses[networkID].GIVING_ADDRESS)) {
```
Exit the function

0xLienid replied:
same question as a few comments ago, doesn't it already? After it console.logs it just jumps straight to the return

0xJem commented on src/helpers/GiveRedemptionBalanceHelper.ts:
```diff
@@ -97,50 +114,60 @@ export const getMockRedemptionBalancesAsync = async ({ address, networkID, provi
+    for (let i = 0; i < events.length; i++) {
+      const event = events[i];
+
+      if (event.topics[2] === zeroPadAddress.toLowerCase()) {
```
Easier to read if you check whether a condition isn't satisfied, and `continue` if so.

0xJem commented on src/helpers/GiveRedemptionBalanceHelper.ts:
```diff
@@ -97,50 +114,60 @@ export const getMockRedemptionBalancesAsync = async ({ address, networkID, provi
+ * @param provider Ethereum network provider object
+ * @returns Number of active donations to the provided address
+ */
 export const getDonorNumbers = async ({ address, networkID, provider }: IBaseAddressAsyncThunk) => {
```
Can we define an explicit return type?

0xJem commented on src/slices/AccountSlice.ts:
```diff
@@ -286,34 +288,49 @@ export const getDonationBalances = createAsyncThunk(
+
+        // ([0x00000...], [0]) is the null return indicating a user has no deposits
+        // This confirms that we did not receive back a null deposit
+        if (allDeposits[0].length != 1 || allDeposits[1][0] != BigNumber.from(0)) {
```
Why not flip this and exit the function if the condition isn't satisfied?

0xLienid replied:
Is there actually a clean way to do this here? The only way I see is flipping it and then copying up the return from the bottom into that if block, but I'm not sure that's more readable/cleaner since we have (identical looking) returns everywhere

0xLienid replied:
probably am just missing something obvious though

0xJem replied:
```
if (allDeposits.length === 0) return;

for (let i = 0; i < allDeposits[0].length; i++) {
   // First entry is of format (etc) if there are no deposits
   if (allDeposits[0][i] == "0x0000 (etc)") break;
   if (allDeposits[1][I].eq(0)) break;

   etc
}
```

Or do the same check out of the for loop:

```
if (allDeposits.length === 0) return;

if (allDeposits[0][0] == "0x0000 (etc)") return;
if (allDeposits[1][0].eq(0)) return;

for (let i = 0; i < allDeposits[0].length; i++) {
```

For the return value, you could define it earlier on so that early returns can use it.

0xJem commented on src/slices/AccountSlice.ts:
```diff
@@ -58,9 +61,7 @@ interface IUserDonationInfo {
-  indexAtLastChange: string;
 }
 
 export const getBalances = createAsyncThunk(
```
Return type?

0xJem commented on src/views/Give/CausesDashboard.tsx:
```diff
@@ -69,11 +69,16 @@ export default function CausesDashboard() {
     } else {
+      // Since the new version of the YieldDirector contract uses IDs as a deposit indexing system
+      // the async thunk used to interact with the contract needs to have an ID passed to it
+      // here we use the ID of "-1" to indicate to the thunk that this is a new deposit and thus
```
Can we extract `-1` into a constant with a meaningful name and use that across the codebase? That also reduces the number of comments that you need to litter everywhere.

0xLienid committed:
Addressing Jem's comments

sam-potter committed:
Abstract ReactQueryProvider

0xJem committed:
Remove hard-coded temporary value

0xJem committed:
Center-align the right-facing arrow

0xJem committed:
Backport text alignment

0xJem committed:
Update snapshot

0xJem committed:
Merge branch 'develop' into tyche-grants

0xJem committed:
Merge branch 'tyche-grants' into give-project-card

sam-potter committed:
Merge branch 'develop' into react-query/wrap-view

sam-potter committed:
Use new DecimalBigNumber helper

0xJem committed:
Merge pull request #1545 from OlympusDAO/give-project-card

Migrate Give screens to Grid component

0xJem committed:
Fix Paper width. Remove extraneous padding around causes/grant dashboards.

0xJem committed:
Apply fullWidth to Paper

0xJem committed:
Merge branch 'develop' into tyche-grants

# Conflicts:
#	src/views/Give/RecipientModal.tsx

0xJem committed:
Better handling of the main Give container at different breakpoints

0xJem committed:
Update snapshots

0xLienid committed:
Adding addresses

0xLienid committed:
merge off of react-query/wrap-view

0xLienid committed:
Writing hooks

0xLienid committed:
Update token guard for Give modal, put in file for getting Give info

0xLienid committed:
Update to using hooks for contract interactions

0xLienid committed:
Adding new query hooks to pull data. Only works for mainnet right now

0xLienid committed:
useDonationInfo hook

0xLienid committed:
Hooks update on network change

0xLienid committed:
Cleaning up files

0xLienid committed:
Fixing components so data refreshes appropriately

0xLienid committed:
Merge with develop

0xLienid committed:
Re-add test snapshots

0xLienid committed:
Fix merge issue in useProtocolMetrics

0xLienid committed:
Documentation

0xLienid committed:
BigNumber, pending transactions, removal of GiveThunk

0xLienid committed:
Fix merge issue on tests

0xLienid committed:
Eliminate excess loading flags for faster data display

0xLienid committed:
Button states, GA tracking

0xLienid committed:
Working through gOHM toggle switch for Give

0xLienid committed:
Unit conversions in and out of gOHM

0xLienid committed:
Ignore deposits less than 1e-9 gOHM, they are rounding errors and can lead to underflow issues

0xLienid committed:
Merge off of develop

0xLienid committed:
Recipient info updates and redeem page asset toggle

0xLienid committed:
Update test snapshots, and add DepositRow tests

0xLienid committed:
More tests (broken right now)

0xLienid committed:
Little bit of styling for toggle

0xLienid committed:
Design tweaks

0xLienid committed:
More tests

0xLienid committed:
Merge develop errors

0xLienid committed:
Update snapshots and add conversion tests

0xLienid committed:
Fix edit deposit error

0xLienid committed:
Fix divide by zero error

0xLienid committed:
Styling adjustments

0xLienid committed:
Extracting helpers, adding a 'before' GA event, minor tweaks

0xLienid committed:
Fix project card test, drop manage modal test for now (visually confirmed it works)

0xLienid committed:
Fix withdrawal error

0xLienid commented on src/components/GiveProject/__tests__/GiveProject.unit.test.tsx:
```diff
@@ -0,0 +1,53 @@
+
+jest.mock("src/hooks/useCurrentIndex");
+
+describe("<ProjectCard />", () => {
```
This is redundant because Brightiron added project card tests in the Give view test set

0xLienid commented on src/helpers/Environment.ts:
```diff
@@ -0,0 +1,268 @@
+ * - Other static methods can be added as needed per
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static
+ */
+export class EnvHelper {
```
idk why this file is listed in my diff when I just merged off of develop...

0xJem replied:
I think this file was removed, and you've probably merged it wrong.

0xLienid commented on .env.example:
```diff
@@ -25,6 +25,14 @@ REACT_APP_FANTOM_TESTNET_NODE_URL=""
 REACT_APP_POLYGON_NODE_URL=""
 REACT_APP_POLYGON_TESTNET_NODE_URL=""
 
+# Olympus Give
```
idk why this file is listed in my diff when I just merged off of develop...

0xJem replied:
I think there were some changes in the interim

0xLienid commented on .env.test:
```diff
@@ -0,0 +1,71 @@
+# .env.test
```
idk why this file is listed in my diff when I just merged off of develop...

brightiron replied:
if this file was unintentionally committed, I'd recommend removing this from this PR, either by removing it, or adding .env.test to .gitignore. This file doesn't exist in develop so it'll be created upon merge.

0xLienid commented on src/helpers/GetCorrectUnits.ts:
```diff
@@ -0,0 +1,39 @@
+import { DecimalBigNumber } from "./DecimalBigNumber/DecimalBigNumber";
+
+export const GetCorrectContractUnits = (
```
Need documentation

0xLienid commented on src/helpers/GetCorrectUnits.ts:
```diff
@@ -0,0 +1,39 @@
+  return convertedValue;
+};
+
+export const GetCorrectStaticUnits = (
```
documentation

0xLienid commented on src/helpers/GetDonationDate.ts:
```diff
@@ -22,23 +41,27 @@ export const GetDonationDate = async ({ address, recipient, networkID, provider
-    const day = date.getDate();
-    const year = date.getFullYear();
+  for (let i = 0; i < events.length; i++) {
+    if (BigNumber.from(events[i].data).gt("0")) firstDonation = events[i];
```
should explain this condition

0xLienid commented on src/views/Give/Give.tsx:
```diff
@@ -96,14 +99,20 @@ function Give({ selectedIndex }: GiveProps) {
 
               <TabPanel value={view} index={0}>
-                <CausesDashboard />
+                <GohmToggle giveAssetType={giveAssetType} changeAssetType={changeAssetType} />
```
The alignment on this with the CausesDashboard container is slightly off and I can't figure out how to fix it without adding custom css...

0xLienid commented on src/views/Give/Give.tsx:
```diff
@@ -96,14 +99,20 @@ function Give({ selectedIndex }: GiveProps) {
               </TabPanel>
               <TabPanel value={view} index={1}>
-                <GrantsDashboard />
+                <GohmToggle giveAssetType={giveAssetType} changeAssetType={changeAssetType} />
```
same as CausesDashboard comment above

0xLienid commented on src/views/Give/RedeemYield.tsx:
```diff
@@ -56,12 +64,15 @@ export default function RedeemYield() {
   const stakingRebasePercentage = new BigNumber(stakingRebase ? stakingRebase : 0).multipliedBy(100);
-  const nextRewardValue = new BigNumber(stakingRebase ? stakingRebase : 0).multipliedBy(totalDeposit);
+  const nextRewardValue = totalDeposit
+    .mul(new DecimalBigNumber(stakingRebase ? stakingRebase.toString() : "0", 2))
```
Does this multiplication look right or should I use stakingRebasePercentage instead?

0xJem replied:
I fixed it in the react-query branch, so when we merge that, it'll be fixed.

0xLienid committed:
Documentation and edits from Jem's old comments

0xLienid committed:
Review tweaks

0xJem committed:
Compilation fix

0xJem committed:
Fix variable dependencies

0xJem committed:
Merge branch 'develop' into give-react-query

# Conflicts:
#	src/components/GiveProject/GrantCard.tsx
#	src/components/GiveProject/ProjectCard.tsx
#	src/locales/translations
#	src/slices/GiveThunk.ts
#	src/views/Give/CausesDashboard.tsx
#	src/views/Give/DepositRow.tsx
#	src/views/Give/GrantsDashboard.tsx
#	src/views/Give/ManageDonationModal.tsx
#	src/views/Give/RecipientModal.tsx
#	src/views/Give/RedeemYield.tsx

0xJem committed:
Remove redundant variables, simplify

0xJem committed:
Fix infinite loop

0xJem committed:
Fix percentage calculation

0xJem committed:
Cleanup of variables

0xJem committed:
Cleanup variables

0xJem committed:
Remove access to state and old cruft

0xJem committed:
Update snapshots (removed skeleton component)

0xJem committed:
Fix incorrect variable used

0xJem commented on src/helpers/GetCorrectUnits.ts:
```diff
@@ -0,0 +1,68 @@
+import { DecimalBigNumber } from "./DecimalBigNumber/DecimalBigNumber";
```
Consider shifting this into a helpers directory underneath Give, or add give to the filename.

0xLienid replied:
Going to move this one into a helpers folder and the helpers that just pull data like GetDonationDate, I'm going to rename with Give included (until they all are moved to a hook for react query)

0xJem commented on src/helpers/GetDonationDate.ts:
```diff
@@ -1,9 +1,28 @@
-import { ethers } from "ethers";
+import { BigNumber, ethers } from "ethers";
```
Same for this. Move, or add give to the name.

0xJem commented on src/helpers/GetDonationDate.ts:
```diff
@@ -1,9 +1,28 @@
+ * @returns String representation of the date as Month Abbreviation DD, YYYY
+ *          i.e. Feb 13, 2022
+ */
+export const GetFirstDonationDate = async ({
```
Why is this using the older `*thunk` style? If all of the fetching is being moved to react-query, shouldn't this be architected in a way to support that?

0xLienid replied:
just because this branch was started a month ago before we started on switching Give to react query and I wasn't sure which would be done first/which the priority would be in getting out first. i.e. didn't want everything to be ready for this but be held up by react query changes

0xJem reviewed (changes requested):
It's hard to complete a review of this. I would suggest that you make the jump in this to use react-query, instead of taking 2 steps forward (react-query) and 1 step backward (leaving the new contract using thunks).

0xJem committed:
Fix bug with skeleton component on project cards

0xJem committed:
Cleanup debt/deposit confusion

0xJem committed:
Rename query with ambiguous name

0xJem committed:
Renaming of ambiguous method

0xJem committed:
Fix bug that resulted a user's donation information not loading

0xJem committed:
Remove redundant items in AccountSlice. Mark other code to be removed.

0xJem committed:
Remove redundant test code

0xJem committed:
Minor console fix

0xLienid committed:
Move awaits out of loops. useDonorNumbers returns a sum of mainnet and testnet though...

0xLienid committed:
Remove console logs, add redeem hook

0xLienid committed:
Update Give test snapshots, and mock returns for redeem yield hooks

0xLienid committed:
Clean up Donation Info hook

0xLienid committed:
Remove redundant files and functions, test updates, modal state updates

0xLienid committed:
Rename/move helper files, fix messed up merge

0xLienid committed:
Merge with develop

0xLienid committed:
Fix merge error

0xJem committed:
Restore redeem amount to modal

0xJem committed:
Cleanup styling in GiveTokenAllowanceGuard. Cleaner use of react-query hook.

0xJem committed:
Restore breakpoint

0xJem committed:
Restore project/grant name to Google Analytics events

0xJem committed:
Renaming artifact

0xJem committed:
Update trivial/styling changes in snapshot

0xJem committed:
Revert snapshot changes

0xJem committed:
Remove redundant code

0xJem committed:
Test fixes for RedeemYield

- Update snapshots with minor styling info
- Update snapshots with metrics that were overwritten
- Shift RedeemYield tests to mock react-query

0xJem committed:
Add test case for 0 redeemable balance

0xJem committed:
Constantly shifting snapshots

0xJem committed:
Merge branch 'develop' into give-react-query

0xJem commented on src/helpers/GiveGetTotalDonated.ts:
```diff
@@ -0,0 +1,126 @@
+ * @returns String representation of the sOHM value of the total amount redeemed by a user
+ *          plus their current redeemable balance
+ */
+export const getTotalDonated = async ({ address, networkID, provider }: IBaseAddressAsyncThunk): Promise<string> => {
```
How is this different (or better) than `useTotalYieldDonated`?

0xLienid committed:
Removing redeeming and mockRedeeming from Account reducer

0xJem committed:
Remove redundant code

0xLienid committed:
Merge with give-react-query

0xLienid committed:
Bug squashing and adding mainnet contract

0xLienid committed:
Update displays and give info hook

0xLienid committed:
Get rid of irrelevant comment

0xJem committed:
Fix translation string

0xJem committed:
Disable button while redeeming

0xJem committed:
Merge branch 'develop' into tyche-new-contract

# Conflicts:
#	src/components/GiveProject/GrantCard.tsx
#	src/components/GiveProject/ProjectCard.tsx
#	src/constants/addresses.ts
#	src/helpers/GiveRedemptionBalanceHelper.ts
#	src/hooks/useGiveInfo.ts
#	src/views/Give/CausesDashboard.tsx
#	src/views/Give/DepositRow.tsx
#	src/views/Give/GrantsDashboard.tsx
#	src/views/Give/Interfaces.ts
#	src/views/Give/ManageDonationModal.tsx
#	src/views/Give/RecipientModal.tsx
#	src/views/Give/RedeemYield.tsx
#	src/views/Give/YieldRecipients.tsx
#	src/views/Give/__tests__/Give.unit.test.jsx
#	src/views/Give/__tests__/RedeemYield.unit.test.jsx
#	src/views/Give/__tests__/__snapshots__/RedeemYield.unit.test.jsx.snap
#	src/views/Give/hooks/useEditGive.ts
#	src/views/Give/hooks/useGive.ts
#	src/views/Give/hooks/useRedeem.ts

0xJem committed:
Fix asset type

0xJem committed:
Add defined type for changeAssetType

0xJem committed:
Port fix for contract allowance

0xJem committed:
Fix regression in number input in RecipientModal. Add tests for it.

0xJem committed:
Refactor interface

0xJem committed:
Replicate number input fix in ManageDonationModal

0xJem committed:
TODO

0xJem committed:
Additional tests for ManageDonationModal (still failing)

0xJem committed:
Add tests

0xJem committed:
4 decimal places

0xJem committed:
Attempting to make it easier to access HTML elements

0xLienid committed:
Cleaning up code

0xLienid committed:
Improve documentation around useGiveInfo return values

0xLienid committed:
Error handling, fix withdrawals

0xLienid committed:
Test updates

0xLienid committed:
Decimals handling and tests

0xJem committed:
Extract 9 decimal places to constant

0xJem committed:
Fix and add tests

0xJem committed:
Decimal places truncation in RecipientModal

0xJem committed:
Truncating to 9 decimal places. Ensure all text reflects the active asset.

0xJem committed:
Accuracy and asset type on RecipientModal

0xJem committed:
Update tests for RedeemYield (currently failing)

0xJem committed:
Adjust values in RedeemYield to be more distinct (with snapshots). Fixed incorrect use of sOHM/gOHM units.

0xJem committed:
Truncate long numbers

0xLienid committed:
Implement withdrawal fix on Grant Card and Project Card

0xJem committed:
Cleanup. Documentation.

0xJem committed:
Merge branch 'develop' into tyche-new-contract

# Conflicts:
#	src/App.tsx
#	src/slices/interfaces.ts
#	src/testHelpers.ts

0xJem committed:
Fix issue with withdrawing

0xLienid committed:
Refresh donations box on withdrawal

0xLienid committed:
Simplify withdrawal to just decimal truncating

0xJem commented on src/hooks/useGiveInfo.ts:
```diff
@@ -435,7 +435,7 @@ export const useDonorNumbers = (address: string) => {
           if (
             potentialActiveDonors[i][0][j].toLowerCase() === address.toLowerCase() &&
-            potentialActiveDonors[i][1][j].gt(0) &&
+            potentialActiveDonors[i][1][j].gt("15000000000") &&
```
Document why this particular number

0xLienid committed:
Merge conflicts

0xLienid committed:
Add hooks to get and redeem old redeemable balance

0xLienid committed:
Redeemable balance on old contract notification and drop down

appleseed-iii commented on .env.example:
```diff
@@ -25,6 +25,14 @@ REACT_APP_FANTOM_TESTNET_NODE_URL=""
+# Olympus Give
+# These 2 settings should be toggled to "true" if you want to use Olympus Give features
+REACT_APP_GIVE_ENABLED="false"
+REACT_APP_GIVE_GRANTS_ENABLED="false"
```
In our prod env we are going to keep this as "false" for now, right?

0xLienid replied:
@0xJem can answer better, I have not been up to date on what the flags should be set to.

0xJem replied:
Bad merge, I think. Should be true in the example file.

Prod also has true values for both, which I think is correct.

appleseed-iii commented on src/App.tsx:
```diff
@@ -125,6 +126,11 @@ function App() {
 
   const { grants } = grantData;
   const { projects } = projectData;
+  const [giveAssetType, setGiveAssetType] = useState<"sOHM" | "gOHM">("sOHM");
+
+  const changeGiveAssetType: ChangeAssetType = (checked: boolean) => {
+    setGiveAssetType(checked ? "gOHM" : "sOHM");
+  };
```
I'm wondering if this should move outside of `app.tsx`. small nit, but would be nice to isolate this inside of the give files. You could put it inside a hook that gets called in the give views... but... that may be messy. lmk what you think.

0xLienid replied:
I agree that it's best for it to not be in `app.tsx`. let me try something to keep it just in the give files and I'll let you know if it's too messy

0xLienid replied:
first thing I tried didn't work smoothly, have one more implementation I can think of that I'll try to do tomorrow

brightiron replied:
I agree this should be refactored. In my brief review it looks like the challenge is that this setting needs to be shared between multiple give routes.

Instead of treating App as the catchall for sharing state between child components, there should be a parent level component for Give that handles this for you. 
We handled a similar case with Wallet by using switch statements to route to the appropriate sub component. We still have routes setup in app.tsx, but we're only pointing to a component.
Wallet index.tsx handles any shared state/prop passing to avoid unnecessary clutter in app.tsx and keep things isolated.

It's additional work, but it'll keep things cleaner in the longrun.

appleseed-iii replied:
ok, as discussed elsewhere. We'll push this refactor into a follow-up next week. @0xLienid

appleseed-iii reviewed (commented):
couple quick comments. will be back to review this more later today.

0xJem commented on src/hooks/useGiveInfo.ts:
```diff
@@ -200,6 +199,49 @@ export const useRedeemableBalance = (address: string) => {
+      queryAssertion([address, networkId], oldRedeemableBalanceQueryKey(address, networkId));
+
+      if (networkId != 1)
+        throw new Error(t`The old Give contract is only supported on the mainnet. Please switch to Ethereum mainnet`);
```
Why not use "V1" instead of old? We have approached it that way with sOHM.

0xJem committed:
Fix test snapshots

0xLienid committed:
Rename interactions from 'old' to 'v1', fix balance showing twice in recipient modal

brightiron commented on src/constants/addresses.ts:
```diff
@@ -62,7 +62,11 @@ export const WETH_ADDRESSES = {
 
 export const GIVE_ADDRESSES = {
-  [NetworkId.TESTNET_RINKEBY]: "0x83D4FE6Ead62547758E094ee5BDb343ADbC2AeB9",
+  [NetworkId.TESTNET_RINKEBY]: "0x5Bb351eeC15A703Afe368f9aa49ccC7de45e5a21",
+  [NetworkId.MAINNET]: "0x42842Fd91c83B84D51D1be514B9716d62c61CbEF",
```
confirmed these are correct contracts

brightiron commented on src/helpers/GiveGetDonationDate.ts:
```diff
@@ -0,0 +1,69 @@
+  const day = date.getDate();
+  const year = date.getFullYear();
+
+  const dateString = month + " " + day + ", " + year;
```
Could you use this here instead?
```suggestion
 const dateString = date.toLocaleDateString('en-US', { year: 'numeric', month: 'short', day: 'numeric' }));
```

brightiron commented on src/networkDetails.ts:
```diff
@@ -45,9 +45,9 @@ export const addresses: IAddresses = {
-    PT_PRIZE_STRATEGY_ADDRESS: "0x2Df17EA8D6B68Ec444c9a698315AfB36425dac8b",
-    GIVING_ADDRESS: "0x83D4FE6Ead62547758E094ee5BDb343ADbC2AeB9",
+    PT_PRIZE_POOL_ADDRESS: "0xf9081132864ed5e4980CFae83bDB122d86619281", // NEW
+    PT_PRIZE_STRATEGY_ADDRESS: "0x2Df17EA8D6B68Ec444c9a698315AfB36425dac8b", // NEW
```
minor, but //new comments were added here

brightiron commented on src/networkDetails.ts:
```diff
@@ -45,9 +45,9 @@ export const addresses: IAddresses = {
-    GIVING_ADDRESS: "0x83D4FE6Ead62547758E094ee5BDb343ADbC2AeB9",
+    PT_PRIZE_POOL_ADDRESS: "0xf9081132864ed5e4980CFae83bDB122d86619281", // NEW
+    PT_PRIZE_STRATEGY_ADDRESS: "0x2Df17EA8D6B68Ec444c9a698315AfB36425dac8b", // NEW
+    GIVING_ADDRESS: "0x5Bb351eeC15A703Afe368f9aa49ccC7de45e5a21",
```
@appleseed-iii  - do these addresses need to be managed in two places? I know you had done some refactor work here.

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/1550/commits/7288cb9bd85b34f646f021cc44796a9ee7fae6d2

brightiron commented:
give snapshot test is also failing here

brightiron committed:
snapshot updates

brightiron commented:
pushed updated snapshots to review diff in test coverage report

0xLienid committed:
Restructure dateString return and delete .env.test

0xLienid committed:
Remove extraneous NEW comments

appleseed-iii commented on src/networkDetails.ts:
```diff
@@ -79,13 +79,13 @@ export const addresses: IAddresses = {
-    PT_PRIZE_POOL_ADDRESS: "0xEaB695A8F5a44f583003A8bC97d677880D528248", // NEW
-    PT_PRIZE_STRATEGY_ADDRESS: "0xf3d253257167c935f8C62A02AEaeBB24c9c5012a", // NEW
+    PT_PRIZE_POOL_ADDRESS: "0xEaB695A8F5a44f583003A8bC97d677880D528248",
+    PT_PRIZE_STRATEGY_ADDRESS: "0xf3d253257167c935f8C62A02AEaeBB24c9c5012a",
```
i'm going to renuke these in a minute, since we don't use them anymore.

appleseed-iii commented on src/networkDetails.ts:
```diff
@@ -79,13 +79,13 @@ export const addresses: IAddresses = {
     TREASURY_V2: "0x9a315bdf513367c0377fb36545857d12e85813ef",
     FIATDAO_WSOHM_ADDRESS: "0xe98ae8cD25CDC06562c29231Db339d17D02Fd486",
-    GIVING_ADDRESS: "0x2604170762A1dD22BB4F96C963043Cd4FC358f18",
+    GIVING_ADDRESS: "0x42842Fd91c83B84D51D1be514B9716d62c61CbEF",
```
I removed these & refactored the one place that they were still used to use the `addresses.ts`... so that we don't have them in 2 places. confusing. We need to clean up this file across the board still.

Anyways, in this change I'm about to push all Tyche addresses are in one spot now, rather than 2.

appleseed-iii reviewed (approved):
good looking out on the test coverage you 👑 s

appleseed-iii committed:
remove extra constant addresses in networkDetails, remove translations (shouldnt exist anymore)

appleseed-iii commented:
last test run failed due to coverage error. tests all passed. merging.
